pre-commit and pre-push git hooks

mnecchi opened the pull request:
# Description

The `pre-commit` has been modified in order to execute prettier and lint checks for each commit.
The tests have been moved to the `pre-push` git hooks.

The git hook checks can be manually disabled calling `git commit` and `git push` with the `--no-verify` flag if needed.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [ ] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

## Checklist

- [ ] I have updated mockServer
- [x] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

Add `npm run test:i18n` to `pre-push` hook?

selvach commented:
@MikeD-ZOE @the-kenneth - Would you be happy with this?

selvach commented:
Hi @mnecchi - What do `git-hooks/pre-commit.js` and `git-hooks/helpers.js` add above what is possible with husky? I'm not very familiar with these tools and so will need to do a bit of digging.

mnecchi commented:
> Hi @mnecchi - What do `git-hooks/pre-commit.js` and `git-hooks/helpers.js` add above what is possible with husky? I'm not very familiar with these tools and so will need to do a bit of digging.

Husky is just a library that makes using git hooks easier.
In Husky you can just say for this git hook run this command or script.
Before you were using Husky to run a simple `npm run test`.
With this PR I've changed the configuration of Husky in the `package.json` to make it run the `git-hooks/pre-commit.js` script for the pre-commit git hook and the `git-hooks/pre-push.sh` script for the pre-push git hook.
`git-hooks/helper.js` is just a collection of functions used by `git-hooks/pre-commit.js` (as I said these are script I wrote for my current job, I just copied, pasted and removed the unnecessary parts).
